lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -214,94 +205,38 @@ def test_get_bad_query_2(self, query_get_as):
 
     def test_put_missing_json_object(self, client, server_config, pbench_token):
         """
-        Test behavior when no JSON payload is given
+        Test behavior when no dataset name is given on the URI. (NOTE that
+        Flask automatically handles this with a NOT_FOUND response.)
         """
-        response = client.put(f"{server_config.rest_uri}/datasets/metadata")
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert response.json.get("message") == "Invalid request payload"
+        response = client.put(f"{server_config.rest_uri}/datasets/metadata/")
```

Comment:
I guess that's a good question, since I had to experiment to answer it; Flask's mapping can be quirky.

In this case, however, it seems to have exactly the same behavior: Flask throws a NOT_FOUND before we get control, as in, "that URL doesn't connect to anything".

I guess I might as well push my new test case...